homeassistant/components/zha/config_flow.py

```diff
@@ -491,16 +550,35 @@ async def async_step_maybe_confirm_ezsp_restore(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Confirm restore for EZSP radios that require permanent IEEE writes."""
-        call_step_2 = await self._radio_mgr.async_restore_backup_step_1()
-        if not call_step_2:
-            return await self._async_create_radio_entry()
 
         if user_input is not None:
-            await self._radio_mgr.async_restore_backup_step_2(
-                user_input[OVERWRITE_COORDINATOR_IEEE]
+            if user_input[OVERWRITE_COORDINATOR_IEEE]:
+                # On confirmation, overwrite destructively
+                await self._radio_mgr.restore_backup(overwrite_ieee=True)
+                return await self._async_create_radio_entry()
+            # On rejection, explain why we can't restore
+            return self.async_abort(
+                reason="cannot_restore_backup",
+                description_placeholders={
+                    "error": "Confirmation not given for IEEE overwrite"
+                },
```

Comment:
Also a minor thing, since the rest of the message (even without the description placeholder) still makes sense, but:

It might be nice if we have a fully custom/translatable message for this case (when the user did not tick the "Permanently replace the radio IEEE address" option).